core/load_balancer_rule.go

```diff
@@ -0,0 +1,184 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type LoadBalancerRuleAlgorithm string
+
+const (
+	RoundRobinRuleAlgorithm LoadBalancerRuleAlgorithm = "round_robin"
+	//nolint:lll
```

Comment:
nitpick: I think ignoring the `lll` linter for this whole const block looks cleaner :)

```suggestion
//nolint:lll
const (
	RoundRobinRuleAlgorithm       LoadBalancerRuleAlgorithm = "round_robin"
```